OSDOCS-9614: Created documentation to create a ROSA with HCP cluster that uses external auth for OIDC (#73839)
openshift/openshift-docs: modules/rosa-hcp-sts-creating-a-cluster-ext-auth-cluster-cli.adoc

// Module included in the following assemblies:
//
// * rosa_hcp/rosa-hcp-sts-creating-a-cluster-quickly.adoc

:_mod-docs-content-type: PROCEDURE
[id="rosa-hcp-sts-creating-a-cluster-external-auth-cluster-cli_{context}"]
= Creating a {hcp-title} cluster that uses external authentication providers
:source-highlighter: pygments
:pygments-style: emacs
:icons: font
 
Use the ROSA CLI's `--external-auth-providers-enabled` flag to create your cluster that uses an external authentication service.

.Procedure

. Use one of the following commands to create your ROSA with HCP cluster:
+

> * If you did not set environmental variables, run the following command:

Author: I need to create an environmental variable module to cover this, but the HCP process uses variables already - https://file.rdu.redhat.com/eponvell/OSDOCS-9614_External-OIDC-Auth-HCP/rosa_hcp/rosa-hcp-sts-creating-a-cluster-quickly.html

> * If you did not set environmental variables, run the following command:

Reviewer: Maybe we can have some examples of the environmental variables?

+
[NOTE]
====
When creating a {hcp-title} cluster, the default machine Classless Inter-Domain Routing (CIDR) is `10.0.0.0/16`. If this does not correspond to the CIDR range for your VPC subnets, add `--machine-cidr <address_block>` to the following commands.
====
+
[source,terminal]
----

> $ rosa create cluster --cluster-name=<cluster_name> \

Reviewer 2: I think this section is reusing the main "how to create a cluster" in 

https://file.rdu.redhat.com/eponvell/OSDOCS-9614_External-OIDC-Auth-HCP/rosa_hcp/rosa-hcp-sts-creating-a-cluster-quickly.html#rosa-hcp-sts-creating-a-cluster-cli_rosa-hcp-sts-creating-a-cluster-quickly

I'd highlight the parameters passed, and at least re-inforce the `--external-auth-providers-enable` flag.

 - Worker:                  arn:aws:iam::<AWS_id>:role/<account_roles_prefix>-HCP-ROSA-Worker-Role
Operator IAM Roles:
 - arn:aws:iam::<AWS_id>:role/<operator_roles_prefix>-openshift-cloud-network-config-controller-clo
 - arn:aws:iam::<AWS_id>:role/<operator_roles_prefix>-kube-system-capa-controller-manager
 - arn:aws:iam::<AWS_id>:role/<operator_roles_prefix>-kube-system-control-plane-operator
 - arn:aws:iam::<AWS_id>:role/<operator_roles_prefix>-kube-system-kms-provider
 - arn:aws:iam::<AWS_id>:role/<operator_roles_prefix>-kube-system-kube-controller-manager
 - arn:aws:iam::<AWS_id>:role/<operator_roles_prefix>-openshift-image-registry-installer-cloud-cred
 - arn:aws:iam::<AWS_id>:role/<operator_roles_prefix>-openshift-ingress-operator-cloud-credentials
 - arn:aws:iam::<AWS_id>:role/<operator_roles_prefix>-openshift-cluster-csi-drivers-ebs-cloud-crede
Managed Policies:           Yes
State:                      ready
Private:                    No
Created:                    Jan 01 2030 14:25:52 UTC
User Workload Monitoring:   Enabled
Details Page:               https://<url>
OIDC Endpoint URL:          https://<endpoint> (Managed)
Audit Log Forwarding:       Disabled

> External Authentication:    Enabled <.>
> ----
> <.> The `External Authentication` flag is enabled, and you can now create an external authentication provider.

Reviewer 3: This [doesn't work on the Customer Portal](https://redhat-internal.slack.com/archives/C02KBD8A4UF/p1675785495598579) :disappointed: 
```suggestion
External Authentication:    Enabled <1>
----
<1> The `External Authentication` flag is enabled, and you can now create an external authentication provider.
```

// Module included in the following assemblies:
//
// * rosa_hcp/rosa-hcp-sts-creating-a-cluster-quickly.adoc

:_mod-docs-content-type: PROCEDURE
[id="rosa-hcp-sts-creating-a-cluster-external-auth-cluster-cli_{context}"]
= Creating a {hcp-title} cluster that uses external authentication providers
:source-highlighter: pygments
:pygments-style: emacs

> :icons: font

Reviewer 3: Is all of this stuff supported on the portal? (I see it looks fine on the preview) :thinking:

> Use the ROSA CLI's `--external-auth-providers-enabled` flag to create your cluster that uses an external authentication service.

Reviewer 3: ```suggestion
Use the `--external-auth-providers-enabled` flag in the ROSA CLI to create a cluster that uses an external authentication service.
```

.Procedure

> . Use one of the following commands to create your ROSA with HCP cluster with the `--external-auth-providers-enabled` flag included:

Reviewer 3: Single step proc:
```suggestion
* Use one of the following commands to create your ROSA with HCP cluster with the `--external-auth-providers-enabled` flag included:
```

Reviewer 3: You could probably slurp the info/note in this step into the intro of the topic, and then just have the two options as steps

 - Worker:                  arn:aws:iam::<AWS_id>:role/<account_roles_prefix>-HCP-ROSA-Worker-Role
Operator IAM Roles:
 - arn:aws:iam::<AWS_id>:role/<operator_roles_prefix>-openshift-cloud-network-config-controller-clo
 - arn:aws:iam::<AWS_id>:role/<operator_roles_prefix>-kube-system-capa-controller-manager
 - arn:aws:iam::<AWS_id>:role/<operator_roles_prefix>-kube-system-control-plane-operator
 - arn:aws:iam::<AWS_id>:role/<operator_roles_prefix>-kube-system-kms-provider
 - arn:aws:iam::<AWS_id>:role/<operator_roles_prefix>-kube-system-kube-controller-manager
 - arn:aws:iam::<AWS_id>:role/<operator_roles_prefix>-openshift-image-registry-installer-cloud-cred
 - arn:aws:iam::<AWS_id>:role/<operator_roles_prefix>-openshift-ingress-operator-cloud-credentials
 - arn:aws:iam::<AWS_id>:role/<operator_roles_prefix>-openshift-cluster-csi-drivers-ebs-cloud-crede
Managed Policies:           Yes
State:                      ready
Private:                    No
Created:                    Jan 01 2030 14:25:52 UTC
User Workload Monitoring:   Enabled
Details Page:               https://<url>
OIDC Endpoint URL:          https://<endpoint> (Managed)
Audit Log Forwarding:       Disabled

> External Authentication:    Enabled

Reviewer 4: I wonder if it would make sense to add a call out in addition to the highlight. Just a suggestion!

Reviewer 3: A callout would be more consistent with the rest of docs - I have never seen the highlighting anywhere. Do we know for sure that it's fully supported by our tooling (mostly, I worry about the portal :sweat_smile:)? Is it ok as far as accessibility is concerned?

----

> . Watch the {hcp-title} installation program logs to track the progress of the cluster creation by running the following command:

Reviewer 3: Is this command telling them to watch, or is it telling them to install while watching?

> . Track the progress of the cluster creation by watching the {hcp-title} installation program logs. To check the logs, run the following command:

Reviewer 4: ```suggestion
.  Watch the {hcp-title} installation program logs to track the progress of the cluster creation by running the following command:
```
Not sure but I think rewording would help with the clarity of this step. It also makes the optional step clearer too I think.
